### PR TITLE
[DEVX-295]: Transfer Learning Notebook

### PR DESCRIPTION
**Pull request by sanjaychelliah**
# what

- Transfer Learn Training Notebook
- Dataset upload functionality change

### CODE REVIEW

File: models/model_train/transfer-learn.ipynb [hunk 1]
```diff
@@ -0,0 +1,614 @@
...
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "id": "d8b58bc6",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import sys\n",
+    "# adding dataset.py folder in food-101 folder to the system path for dataloader object\n",
+    "sys.path.insert(0, module_path)"
```
Comment (sainivedh) [inline models/model_train/transfer-learn.ipynb:224]: we could use `load_module_dataloader` functionality
https://github.com/Clarifai/clarifai-python/blob/4c75dd673615990ce4af5b05a13d0552e1b2402a/clarifai/datasets/upload/utils.py#L14

Just the path of module gives dataloader

File: search/cross_modal_search.ipynb
Comment (sainivedh) [inline search/cross_modal_search.ipynb:1]: There's a small bug at the end
Filter must be
f`ilters=[{"input_dataset_ids": [dataset.id]}]`

**Review by sainivedh — COMMENTED** [review]
Added some comments.

We could add `get_upload_status()` for some of the uploads

File: models/model_train/transfer-learn.ipynb (hunk 1 above)
Comment (sanjaychelliah) [reply models/model_train/transfer-learn.ipynb]: Added those..

File: search/cross_modal_search.ipynb [hunk 2]
```diff
@@ -307,7 +307,7 @@
    "outputs": [],
    "source": [
     "search = app2.search(top_k=2)\n",
-    "response = search.query(ranks=[{\"text_raw\": \"Red pineapples on the beach.\"}], filters=[{\"input_dataset_id\": dataset.id}])"
+    "response = search.query(ranks=[{\"text_raw\": \"Red pineapples on the beach.\"}], filters=[{\"input_dataset_ids\": dataset.id}])"
```
Comment (sainivedh) [inline search/cross_modal_search.ipynb:310]:
```python
filters=[{\"input_dataset_ids\": [dataset.id]}])
```

**Review by sainivedh — APPROVED** [review]
lgtm

**Pull request merged**